src/pages/dar_application/DataAccessRequest_new.js

```diff
@@ -0,0 +1,331 @@
+import { useState, useEffect, useCallback} from 'react';
+import { a, div, h, h4, span, p } from 'react-hyperscript-helpers';
+import isNil from 'lodash/fp/isNil';
+import { DataSet, DAR } from '../../libs/ajax';
+import { FormField, FormFieldTitle, FormFieldTypes, FormValidators } from '../../components/forms/forms';
+
+const searchOntologies = (query, callback) => {
+  let options = [];
+  DAR.getAutoCompleteOT(query).then(
+    items => {
+      options = items.map(function(item) {
+        return item.label;
+      });
+      callback(options);
+    });
+};
+
+const searchDatasets = (query, callback) => {
+  DataSet.searchDatasets(query).then(items => {
+    let options = items.map(function (item) {
+      return formatSearchDataset(item);
+    });
+    callback(options);
+  });
+};
+
+const formatSearchDataset = (ds) => {
+
+  return {
+    key: ds.dataSetId,
+    value: ds.dataSetId,
+    dataset: ds,
+    displayText: ds.datasetIdentifier,
+    label: span({}, [
+      span({style: {fontWeight: 'bold'}}, [ds.datasetIdentifier]), ' | ', ds.name]),
+  };
+};
+
+const datasetsContainDataUseFlag = (datasets, flag) => {
+  return datasets.some((ds) => ds?.dataUse[flag] === true);
+
+};
+
+const fetchAllDatasets = async (dsIds) => {
+  let datasets;
+  if (!isNil(dsIds)) {
+    datasets = await Promise.all(dsIds.map((id) => DataSet.getDatasetByIdV2(id)));
+  } else {
+    datasets = [];
+  }
+
+  return datasets;
+};
+
+//NOTE: need to change props to account for file locations for previous uploaded file
+export default function DataAccessRequest(props) {
+  const {
+    formFieldChange,
+    formData,
+    setIrbDocument,
+    setCollaborationLetter,
+    ariaLevel = 2
+  } = props;
+
+  const [datasets, setDatasets] = useState([]);
+  const today = new Date();
+  const irbProtocolExpiration = formData.irbProtocolExpiration || `${today.getFullYear().toString().padStart(4, '0')}-${today.getMonth().toString().padStart(2, '0')}-${today.getDate().toString().padStart(2, '0')}`;
```

Comment:
is the irb expiration always one year from the time of upload?

Reply:
Good question. @solideoglori, any insights?